flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergStreamWriterMetrics.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.sink;
+
+import com.codahale.metrics.SlidingWindowReservoir;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.flink.dropwizard.metrics.DropwizardHistogramWrapper;
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.Histogram;
+import org.apache.flink.metrics.MetricGroup;
+import org.apache.iceberg.io.WriteResult;
+
+class IcebergStreamWriterMetrics {
+  // 1,024 reservoir size should cost about 8KB, which is quite small.
+  // It should also produce good accuracy for histogram distribution (like percentiles).
+  private static final int HISTOGRAM_RESERVOIR_SIZE = 1024;
+
+  private final Counter flushedDataFiles;
+  private final Counter flushedDeleteFiles;
+  private final Counter flushedReferencedDataFiles;
+  private final AtomicLong lastFlushDurationMs;
+  private final Histogram dataFilesSizeHistogram;
+  private final Histogram deleteFilesSizeHistogram;
+
+  IcebergStreamWriterMetrics(MetricGroup metrics, String fullTableName) {
+    MetricGroup writerMetrics = metrics.addGroup("IcebergStreamWriter", fullTableName);
```

Comment:
The group is equivalent to a tag for a tag based reporter in Flink (e.g. Prometheus). Would something like `table` be clearer name for users who want to aggregate metrics by the full table name? 

You can still use `IcebergStreamWriter` as a group without a value as a means to identify that metric belongs to `IcebergStreamWriter` in addition to the suggestion

Reply:
agree. Thanks for your suggestions!